### DISCUSSION TITLE
DBC Muxed Values

### DISCUSSION THREAD

**Discussion by MichaelBMiner** [cantools/cantools#453] [General]
Hello,

I am looking for some advice on how to create a DBC entry for data that must be muxed. 
![MuxedStuff](https://user-images.githubusercontent.com/7850720/178339623-41140f08-0192-4300-9fe2-45b9ef06e1aa.PNG)

When writing my DBC entry do I assume the first three values are the ones to be controlled by a mux? 

This data is sent from an EVCC to an EV on request. When the EV requests this data will it provide a mux identifier so that I can properly populate the data? Obviously this would be implementation specific, but this would make the most sense to me.

**Comment by juleq** [comment]
Thats J1939. In J1939 there is a multiplex method at work that does use the extended CAN identifier. It is well formatted and its components like SPN and PGN specify the payload contents.

Some data will be broadcasted, some is sent on request. Here some info on such request:

`To send a J1939 request via the CAN bus, a special 'request message' is used (PGN 59904), which is the only J1939 message with only 3 bytes of data. It has priority 6, a variable transmit rate and can either be sent as a global or specific address request. The data bytes 1-3 should contain the requested PGN (Intel byte order). Examples of requested J1939 messages include the diagnostic messages (e.g. J1939 DM2). `

Source: https://www.csselectronics.com/pages/j1939-explained-simple-intro-tutorial
